lib/dal/src/component/debug.rs

```diff
@@ -0,0 +1,177 @@
+}
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum ComponentDebugViewError {
+    #[error(transparent)]
+    AttributeDebugViewError(#[from] AttributeDebugViewError),
+    #[error(transparent)]
+    AttributeValue(#[from] AttributeValueError),
+    #[error("Attribute Value tree badly constructed with root prop of {0}")]
+    AttributeValueTreeBad(AttributeValueId),
+    #[error("component error: {0}")]
+    Component(String),
+    #[error(transparent)]
+    ComponentError(#[from] ComponentError),
+    #[error(transparent)]
+    Func(#[from] FuncError),
+    #[error(transparent)]
+    FuncExecution(#[from] FuncExecutionError),
+    #[error("json pointer not found: {1:?} at {0}")]
+    JSONPointerNotFound(serde_json::Value, String),
+    #[error(transparent)]
+    NodeWeightError(#[from] NodeWeightError),
+    #[error("no internal provider for prop {0}")]
+    NoInternalProvider(PropId),
+    #[error("no root prop found for schema variant {0}")]
+    NoRootProp(SchemaVariantId),
+    #[error("schema variant not found for component {0}")]
+    NoSchemaVariant(ComponentId),
+    #[error("component not found {0}")]
+    NotFound(ComponentId),
+    #[error(transparent)]
+    Prop(#[from] PropError),
+    #[error(transparent)]
+    SchemaVariant(#[from] SchemaVariantError),
+    #[error(transparent)]
+    Secret(#[from] SecretError),
+    #[error("secret not found: {0}")]
+    SecretNotFound(SecretId),
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+    #[error(transparent)]
+    UlidDecode(#[from] ulid::DecodeError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
+
+impl ComponentDebugView {
+    #[instrument(level = "info", skip_all)]
+    pub async fn new(ctx: &DalContext, component: &Component) -> ComponentDebugViewResult<Self> {
+        let schema_variant = Component::schema_variant(component, ctx).await?;
+
+        let child_values =
+            Self::get_attribute_value_tree_for_component(ctx, component.id()).await?;
+
+        let name = component
+            .name(ctx)
+            .await
+            .map_err(|e| ComponentDebugViewError::Component(format!("get name error: {}", e)))?;
+
+        let debug_view = ComponentDebugView {
+            name,
+            schema_variant_id: schema_variant.id(),
+            attribute_tree: child_values,
+        };
+
+        Ok(debug_view)
+    }
+    #[instrument(level = "info", skip_all)]
+    pub async fn get_attribute_value_tree_for_component(
+        ctx: &DalContext,
+        component_id: ComponentId,
+    ) -> ComponentDebugViewResult<HashMap<AttributeValueId, Vec<AttributeValueId>>> {
+        let mut child_values = HashMap::new();
+        // Get the root attribute value and load it into the work queue.
+        let root_attribute_value_id = Component::root_attribute_value_id(ctx, component_id).await?;
+        let workspace_snapshot = ctx.workspace_snapshot()?;
```

Comment:
I'm a bit conflicted. I see other `AttributeValue` methods called in here, so that's a sign to create more methods to sandbox the snapshot. However, it makes more sense in this function than before because this is more of a raw usage scenario. I'd still investigate breaking them up.